### ISSUE TITLE
parameters with period do not work

### ISSUE THREAD

**Issue by ghost** [jacwright/RestServer#5]
I have a method like:

``` php
/**
* Returns a parameter
*
* @url POST /parameter/$parameter
*/
public function parameter($parameter = null)
{
return $parameter;
}
```

My request is /parameter/12334556.pdf

From above method I get 12334556 and should be 12334556.pdf

**Comment by ningt** [comment]
Isn't removing period afterwards the desired behaviour? @jacwright 

In `getPath` method:

    // remove trailing format definition, like /controller/action.json -> /controller/action
    $path = preg_replace('/\.(\w+)$/i', '', $path);

**Comment by designstorming** [comment]
For me, it removed **.com** from the end of the email string. I just appended **.something** after and got the whole email address

**Comment by jacwright** [comment]
This could be something where we remove the period for matching the action, but include it for the action arguments as described in the situation above. If someone would like to contribute a fix back that can handle both situations I would be happy to merge it in.

**Comment by tyzoid** [comment]
This requires some careful thought. My first instinct is to change the character that deliminates the format string or make it configurable. This would still be in the brainstorming stage.

**Comment by tyzoid** [comment]
I submitted a fix, currently in the bugfix/#5 branch, which only removes the extension when a defined format is matched. This is more of a workaround, a true fix would not modify the urls except for proper routing. Later I'll also include a flag that will disable this format switching by extension, and rely solely on the Accepts header.

Can someone verify that this resolves the issue with the caveat above?
